src/main/java/uk/ac/ebi/subs/apisupport/ApiSupportService.java

```diff
@@ -72,7 +73,7 @@ public void deleteSubmissionContents(Submission submission) {
     public void markContentsAsSubmitted(uk.ac.ebi.subs.data.Submission submission) {
 
         final String submissionId = submission.getId();
-        Submission currentSubmission = submissionRepository.findById(submissionId)
+        Submission currentSubmission = Optional.of(submissionRepository.findOne(submissionId))
```

Comment:
please use Optional.ofNullable(). otherwise the api throws an error if the passed argument is null. i experienced this myself.

Reply:
If I would apply your change then the next line could throw a NPE